accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,199 @@
+   - In the HUB, generate authentication tokens to access the channel(s). Re-use the existing channel access table.
+   - In the peripheral create all needed channels (vendor and custom) point to the HUB server in the rhncontentsource table. Each channel needs to have its own entry.
+   - Minimal set of tables will be populated on the peripheral: rhnchannel,  rhnchannelcontentsource, rhnchannelproduct, rhnchannelfamilymembers, rhnproductname, rhnchannelcloned, suseproductchannel.
+
+
+When creating the channels in the peripheral server the content source will reference the correct HUB repository that provides that channel. Peripheral servers need to be configured with the flag `java.unify_custom_channel_management`, which will synchronize custom channels during the nightly mgr-sync process.
+
+One important aspect is to recreate the connection between custom channels and vendor channels, so we can have SP migration and avoid the need to synchronize all channel clone chains (ISSv1 also does this implementation).
+
+Vendor channels are not linked to any organization, and can also be synchronized with this method.
+
+## Communication Workflow
+
+![Communication workflow](images/hub_sync_content/comunication_workflow.png)
+
+## Integration with SUSE Manager 5.0
+
+With SUSE Manager containerization we cannot register the peripheral servers as minions anymore. Several components were relying on having that.
+In this section we will have a walk-through of those problems and how this solution can help solve them.
+
+### HUB XML-RPC API
+
+Component responsible for broadcasting API requests to peripheral servers. This component loads the peripheral servers that are registered on the HUB by looking at the minion entitlements.
+
+With the proposed solution we can retrieve the list of the registered peripheral servers and avoid the peripheral server registration as minion.
+
+### Inter-server-sync-v2
+
+Will be replaced by this solution. It will continue to exist but for disconnect set-ups only.
+
+### Uyuni-config formula
+
+This proposal doesn't address any of the issues with the configuration formula.
+
+### HUB Report Federation
+
+Reporting database information is collected using a salt state to get the report DB connection information.
+With this solution we can replace the mechanism by an API call to the peripheral server which would return the needed information, and in this way remove the need to minion registration.
+
+
+## Future Steps
+
+This implementation sets the foundations to add more online content synchronization through the API, like configuration channels, images and activation keys.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Full synchronize a high number of peripherals at same time can overload the HUB server
+  - After the first full synchronization it should not be a problem
+  - One Peripheral can manage up to 10.000 minions. If we add 10 peripherals we would be able to manage 100.000 minions (more than our biggest customer)
```

Comment:
This item is not really a drawback to me and doesn't change over ISSv2 architecture

Reply:
In ISSv2 if one adds more peripherals it will not reflect in more HUB load since the export is done once and the import is done in parallel in each peripheral. 
With this approach, adding one more peripheral will add load to the HUB server